src/main/java/com/googlecode/lanterna/gui2/AnimatedLabel.java

```diff
@@ -164,9 +164,8 @@ private AnimationTimerTask(AnimatedLabel label) {
         @Override
         public void run() {
             AnimatedLabel animatedLabel = labelRef.get();
-            if(animatedLabel == null) {
-                cancel();
-                canCloseTimer();
+            if(animatedLabel == null || animatedLabel.getTextGUI() == null) {
```

Comment:
If WaitingDialog::createDialog and WaitingDialog::showDialog have sufficient time between these two calls, the AnimatedLabel will never begin as its textGui is null when the AnimatedLabel starts